docs/cookbook/deploying-to-heroku.md

```diff
@@ -0,0 +1,222 @@
+
+```bash
+$ heroku git:remote -a <yourAppName>
+```
+
+You can check to make sure this worked by using the command:
+
+```bash
+$ git remote -v
+```
+You should see output like, but with your app name at the end:
+
+```bash
+heroku  https://githeroku.com/apos-heroku-tut.git (fetch)
+heroku  https://githeroku.com/apos-heroku-tut.git (push)
+```
+
+## Add a MongoDB Atlas cluster
+With our project set-up to deploy to Heroku, we need a database.
+
+Heroku runs our node app, but it doesn't run MongoDB for us. So let's go to [MongoDB Atlas](https://www.mongodb.com/cloud/atlas) and sign up.
+
+Once you log-in to your account, create a project. I named my project 'apostrophe', but you can use any name you wish. Next, within that project, click on 'Build a Database'. Select the free'Shared' tier and pick 'AWS' as your cloud provider and the same region you chose for Heroku.
+For "Cluster Tier" choose "M0 Sandbox".
+
+We recommend you give your cluster the same name as your project, in the case of this example, 'heroku-tut'.
+
+You will need to set up an administrative MongoDB user for your cluster. These will be part of your MongoDB database connection credentials. **Be sure to set a secure, separate username and password,** do not use your Atlas login credentials. One thing to note: the characters `: / ? ! # [ ] @` in either your username or password will need to be [URL-encoded](https://www.eso.org/~ndelmott/url_encode.html) in the database connection string.
+
+### IP address allowing
+
+MongoDB Atlas requires us to allowlist the IP addresses that should be allowed to talk to the database. **Yes, it is secured by an encrypted password,** but this still helps to reduce potential DOS attacks.
+
+This is a problem with Heroku because it may connect from many IP addresses.
+
+If you are buying a larger Atlas plan you may be able to use the "VPC Peering" option, the details of which are beyond the scope of this recipe. Otherwise, use this IP address range:
+
+`0.0.0.0/0`
+
+This will allow you to connect for the purposes of this tutorial, but should be modified for a production site. One solution is to use the [fixie-socks](https://elements.heroku.com/addons/fixie-socks) add-on from Heroku.
+
+### Telling Heroku about your database
+
+You will need to set an environment variable in Heroku so that your dynos can communicate with the database.Within your new database deployment, click on the 'connect' button next to the name. Next, click on 'Connect your application'. By default, the driver drop-down menu should be set to 'Node.js' and version '4.1 or later'. If they aren't, change thhem and then copy the connection string. Paste the string into a text editor and add your password. Again, make sure to URL encode any special special characters. Now use this string to set set a environment variable in heroku. It should look something like this:
```

Comment:
database.Within -> database. Within

Reply:
tthem -> them

Reply:
"special special" is very special 😄

Reply:
Changed along with 'set set' the set of all sets!